core/src/main/java/org/mobilitydata/gtfsvalidator/notice/Notice.java

```diff
@@ -40,11 +42,13 @@ public SeverityLevel getSeverityLevel() {
   }
 
   /**
-   * Returns a descriptive type-specific name for this notice.
+   * Returns a descriptive type-specific name for this notice based on the class simple name.
    *
    * @return notice code, e.g., "foreign_key_error".
    */
-  public abstract String getCode();
+  public String getCode() {
+    return StringUtils.removeEnd(getClass().getSimpleName(), NOTICE_SUFFIX);
```

Comment:
Class name is UpperCamelCase, so it does not have "_notice" suffix. I think that you need to use CaseFormat. A unit test would be a great way to catch possible bugs here.

Reply:
Fixed in https://github.com/MobilityData/gtfs-validator/pull/788/commits/eecd38979e57f5bf5ad8d5db91356520aa975bc4.